frontend/src/components/form/components/StateMachine.vue

```diff
@@ -170,9 +173,25 @@ export default {
             }
             this.nextStates.push({state: currentStateItem.value, by: []})
 
+            
+            let ancestors = []
+            if (this.orgName){
+                ancestors = this.ancestorsByName(this.orgName);
+            }
+
             let sysAdmin = this.sysAdmin;
-            let admin = (this.userPermissions[this.orgName] === "admin") 
+            let admin = (this.userPermissions[this.orgName] === "admin");
+            if (!admin){
+                for (let i=0; i<ancestors.length; i++){
+                    admin = ( (admin) || (this.userPermissions[ancestors[i]] === "admin") );
+                }
+            }
```

Comment:
Wrapping the loop in a conditional is not necessary on account of `(admin) || ...`. A case could certainly be made to keep it for readability's sake, however.

Here's another approach to consider. It condenses the logic, and I find it easier to follow.

```javascript
let admin = [this.orgName, ...ancestors].some(
  orgName => this.userPermissions[orgName] === "admin"
);
```

Reply:
We've moved this to a mixin and have moved to the some functionality as suggested